### PR TITLE
Sonar / CPE updates

### PR DESCRIPTION
**Pull request by tsellers-r7**
## Description
This PR:

1. Adds coverage for a few new services
1. Adds/adjusts `service.vendor` and/or `service.product` where necessary to enable CPE auto-generation using NIST issued CPE values. This **significantly** improves the number of services on the Internet that we will have CPEs for. For example, from recent Project Sonar FTP studies the following fingerprints did not have CPEs:

```
Count       Fingerprint description text
2,945,424   "Pure-FTPd versions >= 1.0.14 - Config data can be zero or more of: [privsep] [TLS]"
  625,091   "vsFTPd (Very Secure FTP Daemon)"
   20,529   "Serv-U Serv-U with version on Windows"
    6,224   "vsFTPd (Very Secure FTP Daemon) error message"
    5,104   "Basic Pure-FTPd banner, no version"
      764   "vsFTPd (Very Secure FTP Daemon) extended build (vsftpd.devnet.ru)"
      103   "vsFTPd (Very Secure FTP Daemon) - break me variant"
```

For `53/udp` DNS data, metrics on fingerprinted services that did not have CPE values:
```
Count     Fingerprint description text
986,657  "NLnet Labs Name Server Daemon"
 16,201  "NLnet Labs Unbound"
  2,016  "Knot DNS"
    990  "NLnet Labs Unbound no version string"
```

For `25/tcp` SMTP data, metrics on fingerprinted services that did not have CPE values:
```
Count      Fingerprint description text
  933,600  "Postfix - generic banner"
  315,820  "Postfix - Ubuntu"
  267,531  "Postfix - Debian"
  138,611  "Sendmail - optional timezone and timestamp, w/o OS"
  110,246  "Sendmail - short banner w/o hostname, version, platform, or date."
   13,737  "Postfix - banner without hostname or version"
   11,733  "Postfix - generic banner with amusing comments in parentheses"
    7,888  "Sendmail - with date, w/o version or platform, optional status string."
    7,590  "Postfix - Ubuntu, Mail-in-a-Box package"
    6,816  "Sendmail - Debian 7.x (wheezy)"
    4,766  "Sendmail - with timezone and timestamp, w/o timezone offset or OS"
    3,178  "Postfix - Std semantic versioning, w/ optional parens"
    3,167  "IBM Domino SMTP MTA"
    3,145  "Communigate Pro"
    2,703  "Sendmail - Debian patch only"
```

For `110/tcp` POP3 data, metrics on fingerprinted services that did not have CPE values:
```
Count      Fingerprint description text
3,092,531  "Dovecot Secure POP Server"
```

For `143/tcp` IMAP data, metrics on fingerprinted services that did not have CPE values:
```
Count      Fingerprint description text
3,104,118 "Dovecot Secure IMAP Server"
  308,033 "Courier MTA IMAP"
   14,883 "CMU Cyrus IMAP"
      143 "CMU Cyrus IMAP on Mac OS X"
```

## Motivation and Context
Improved coverage

## How Has This Been Tested?
`rspec`, local testing.

## Types of changes
Content update

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

File: xml/ftp_banners.xml [hunk 1]
```diff
@@ -463,6 +505,7 @@ more text</example>
     <example service.version="0.9.13a beta">FileZilla Server version 0.9.13a beta</example>
     <example service.version="0.9.54 beta">FileZilla Server 0.9.54 beta</example>
     <example service.version="0.9.33 beta">FileZilla Server v0.9.33 beta</example>
+    <param pos="0" name="service.vendor" value="Filezilla-Project"/>
```
Comment (tsellers-r7) [inline xml/ftp_banners.xml:508]: There is no NIST issued CPE value for the Filezilla FTP server.  This change just aligns with vendor value with the vendor in the existing NIST CPEs for the client.

File: xml/http_servers.xml [hunk 2]
```diff
@@ -1151,18 +1152,29 @@
   <fingerprint pattern="^Sun GlassFish Enterprise Server v(\S+)$">
     <description>Glassfish with version information</description>
     <example service.version="2.1">Sun GlassFish Enterprise Server v2.1</example>
-    <param pos="0" name="service.vendor" value="Sun"/>
+    <param pos="0" name="service.vendor" value="Oracle"/>
```
Comment (tsellers-r7) [inline xml/http_servers.xml:1155]: All CPEs issued for Glassfish use `oracle` as the vendor:
https://nvd.nist.gov/products/cpe/search/results?keyword=glassfish&status=FINAL&orderBy=CPEURI&namingFormat=2.3

File: xml/ftp_banners.xml [hunk 3]
```diff
@@ -463,9 +505,14 @@ more text</example>
...
     <param pos="0" name="service.family" value="FileZilla FTP Server"/>
     <param pos="0" name="service.product" value="FileZilla FTP Server"/>
     <param pos="1" name="service.version"/>
+    <param pos="0" name="os.vendor" value="Microsoft"/>
+    <param pos="0" name="os.family" value="Windows"/>
+    <param pos="0" name="os.product" value="Windows"/>
+    <param pos="0" name="os.cpe23" value="cpe:/o:microsoft:windows:-"/>
```
Comment (tsellers-r7) [inline xml/ftp_banners.xml:515]: While the Filezilla **client** is cross platform, the server is **Windows only**.  

https://filezilla-project.org/index.php

Comment (brudis-r7) [reply xml/ftp_banners.xml:515]: It's also a pretty terrible client/server combo :-)

**Comment by tsellers-r7** [comment]
Sorry for the late additions and dismissing the reviews. I found that the NLnet Labs DNS changes I just landed added CPEs for over a million matches from our recent DNS study and so I wanted to get it in now.

File: xml/dns_versionbind.xml [hunk 4]
```diff
@@ -516,6 +516,7 @@
     <param pos="0" name="service.family" value="NSD"/>
     <param pos="0" name="service.product" value="dnsd"/>
     <param pos="1" name="service.version"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:nlnetlabs:name_server_daemon:{service.version}"/>
```
Comment (tsellers-r7) [inline xml/dns_versionbind.xml:519]: This change alone, via the cpe-remap change, added CPEs for over a million matches from our recent DNS study and so I wanted to get it in now.

**Pull request merged**